x/blob/types/params.go

```diff
@@ -90,6 +90,10 @@ func validateGovMaxSquareSize(v interface{}) error {
 			govMaxSquareSize,
 		)
 	}
+	upperBound := uint64(appconsts.SquareSizeUpperBound(appconsts.LatestVersion))
```

Comment:
@cmwaters is it concerning that this line always uses `appconsts.LatestVersion`? 

I think this means that when celestia-app cuts a new major version, the validation of this governance param _may_ change which seems like breaking behavior but that's fine because it will only happen in a breaking release.

What seems fishy is that this function now relies on a global const rather than plumbing app version for a particular block into here.

Reply:
[blocking]

I believe this is the reason why we are not doing this in the first place. We have to route the app version only when it changes if we want single binary syncs. Otherwise the node will change this value whenever they upgrade which will result in non-determinism

Reply:
Good point seems very sketchy. I still think it would be valuable to enforce that the gov max square size is always less than the upper bound constant but I don't think the approach in this PR is safe b/c the reason you mentioned.

Reply:
It is difficult to reject blobs that are too large considering (https://github.com/celestiaorg/celestia-app/pull/2202) because it isn't obvious how to fetch the max square size.